### PR TITLE
Redesigned users tests

### LINKED ISSUE #29: Redesign users.tests with pytest
Following @Omeramsc request for changes in the "Adding base user" PR:
The tests have to be redesigned to use `fixture` or `parametrize` or both!
links:
https://docs.pytest.org/en/stable/fixture.html
https://docs.pytest.org/en/stable/parametrize.html

### PR DESCRIPTION
**Pull request by Micha546**
Used fixture and parametrize from pytest to write better tests!
**note:** I Did not add new tests, Only redesigned the existing ones

links I used:
https://djangostars.com/blog/django-pytest-testing/
https://docs.pytest.org/en/stable/fixture.html
https://docs.pytest.org/en/stable/assert.html

Fix #29

### CODE REVIEW

**Review by simzacks — APPROVED** [review]

File: users/tests.py
Comment (Omeramsc) [inline users/tests.py]: instead of asserting `superuser_model.is_active is True`, you can just `assert super_user_model.is_active`.
same for all the other boolean assertions.

File: users/tests.py
Comment (Omeramsc) [inline users/tests.py]: same as before

File: users/tests.py
Comment (Omeramsc) [inline users/tests.py]: is the case of False, `assert not ... `

**Review by Omeramsc — CHANGES_REQUESTED** [review]
Good job! Great improvements. :) 
I left some tiny comments, fix them and this PR will be good to go.

File: users/tests.py
Comment (Micha546) [reply users/tests.py]: Done, waiting for your approval 💯

**Comment by OrAsaf** [comment]
LGTM

**Review by OrAsaf — APPROVED** [review]

**Review by Omeramsc — APPROVED** [review]
LGTM, Great job!

**Pull request merged**
